### ISSUE TITLE
L2 / R2 as rear touch

### ISSUE THREAD

**Issue by mautheus** [xerpi/ds4vita#59]
In PS Vita games the rear touch is not compatible with DS4. Would it be possible to use the L2 / R2 as a rear touch? So PS Vita games (as God of War Collection uses rear touch) would be great for playing with the DS4

**Comment by xerpi** [comment]
Yes, that would be technically possible but currently, I lack the time to do that. If you know C I can tell you how to do that.

**Comment by mautheus** [comment]
Unfortunately I have no programming knowledge. I also thought of, if possible, the L2 match the left side of the rear touch and the R2 on the right side. I'm sorry I can't help with the programming of the code.

Enviada do Outlook Mobile<https://aka.ms/blhgte>
________________________________

**Comment by TheIronUniverse** [comment]
I would recommend using remaPSV.suprx, it works really well for mapping touch zones to DS4 on a per game basis. you'll have to use a custom release of it, I have it if u need it. :)

**Comment by mautheus** [comment]
I would be very grateful if you can provide me with this plugin. Because it's just missing to be perfect the use of DS4 on PS Vita.

Enviada do Outlook Mobile<https://aka.ms/blhgte>

________________________________

**Comment by mautheus** [comment]
So I tried to map the rear touch with the remaPSV but couldn't. Remember that I'm kind of doing using a DS4. I believe it will be perfect when xerpi can map using its own ds4vita plugin.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by fxb6476** [comment]
@xerpi I know C/C++ explain to me what we need to do. I got some free time this week/weekend and can try to bang out a prototype.

**Comment by xerpi** [comment]
Check this fork: https://github.com/KyleBeebe/ds4vita/tree/hooks

**Comment by fxb6476** [comment]
So it seems that your new-ish "hook" branch has the support for mapping L2 and R2 ala KyleBeebe's solution. I'll test out compiling your latest hook branch and see if it works with moonlight "specifically" what I am concerned with.

Also found that this solution works and is really nice with its pop-up UI. Maybe we should integrate a similar UI experience into ds4? -> https://github.com/MERLev/remaPSV2

**Comment by HRudyPlayZ** [comment]
i feel like this will have a lot of incompatibilies... and could be annoying for other uses :P
To be honest, i prefer the remaPSV2 implementation as it gives us more freedom and still keeps compatibility with the real L2/R2 keys if needed.

**Comment by KyleBeebe** [comment]
> i feel like this will have a lot of incompatibilies... and could be annoying for other uses :P
> To be honest, i prefer the remaPSV2 implementation as it gives us more freedom and still keeps compatibility with the real L2/R2 keys if needed.

You are probably right. The L2/R2 thing was going to be temporary until I added remap functionality but between remaPSV2 and MiniVitaTV existing already, I don't really see a point.

**Comment by mautheus** [comment]
I'm using RemaPSV2 and it looks great. Thank you all so much for the information.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by johnnyq3** [comment]
> > i feel like this will have a lot of incompatibilies... and could be annoying for other uses :P
> > To be honest, i prefer the remaPSV2 implementation as it gives us more freedom and still keeps compatibility with the real L2/R2 keys if needed.
> 
> You are probably right. The L2/R2 thing was going to be temporary until I added remap functionality but between remaPSV2 and MiniVitaTV existing already, I don't really see a point.

I really liked having a plugin that I didn't have to mess around with settings and it just worked.  I understand where you are coming from though and thank you for the work you put in this great plug-in.

**Comment by xerpi** [comment]
Check: https://github.com/MERLev/ds34vita
